### PR TITLE
Feature/1001 avoid segfault on empty literal mergestring

### LINKED ISSUE #1001: Segmentation Fault on "#{}:"
I got Segmentation-Fault from activerecord code and found minimum reproducing code:
https://github.com/rails/rails/blob/master/activerecord/lib/active_record/inheritance.rb#L194

Ruby version: `ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin18]`
Sorbet version: `Sorbet typechecker 0.4.4289 git 91b825ede856537064244202b688317bb00105ac built on 2019-06-21 23:19:41 GMT with debug symbols`
Gemfile.lock:

```
GEM
  remote: https://rubygems.org/
  specs:
    sorbet (0.4.4289)
      sorbet-static (= 0.4.4289)
    sorbet-runtime (0.4.4289)
    sorbet-static (0.4.4289-universal-darwin-14)

PLATFORMS
  ruby

DEPENDENCIES
  sorbet
  sorbet-runtime

BUNDLED WITH
   1.16.2
```

Minimum Segmentation-fault reproducing input:

```
"#{}:"
```

Output:
```
/Users/pocket7878/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/sorbet-0.4.4289/bin/srb: line 43: 21015 Segmentation fault: 11  "${sorbet[0]}" "${args[@]}"
```

### PR DESCRIPTION
**Pull request by pocket7878**
### Motivation

close #1001 

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by DarkDimius** [comment]
Thank you for great contribution and great testcase!

File: ast/desugar/Desugar.cc [hunk 1]
```diff
@@ -94,7 +94,10 @@ unique_ptr<Expression> mergeStrings(DesugarContext dctx, core::Loc loc,
             loc,
             dctx.ctx.state.enterNameUTF8(fmt::format(
                 "{}", fmt::map_join(stringsAccumulated.begin(), stringsAccumulated.end(), "", [&](const auto &expr) {
-                    return cast_tree<Literal>(expr.get())->asString(dctx.ctx).data(dctx.ctx)->shortName(dctx.ctx);
+                    if (isa_tree<EmptyTree>(expr.get()))
+                        return std::basic_string_view("");
```
Comment (DarkDimius) [inline ast/desugar/Desugar.cc]: `""sv` will make it more readable.

File: ast/desugar/test/desugar_test.cc [hunk 2]
```diff
@@ -30,3 +30,14 @@ TEST(DesugarTest, SimpleDesugar) { // NOLINT
...
+TEST(DesugarTest, EmptyTreeDesugar) { // NOLINT
+    sorbet::core::GlobalState gs(errorQueue);
+    gs.initEmpty();
+    sorbet::core::UnfreezeNameTable nameTableAccess(gs);
+    sorbet::core::UnfreezeFileTable ft(gs);
+
+    auto ast = sorbet::parser::Parser::run(gs, "<test>", "\"#{}:\"");
+    sorbet::core::MutableContext ctx(gs, sorbet::core::Symbols::root());
+    auto o1 = sorbet::ast::desugar::node2Tree(ctx, move(ast));
+}
```
Comment (jez) [inline ast/desugar/test/desugar_test.cc]: This test case is redundant with the `test/testdata` test, and the other one is shorter and easier to grep for.

Do you mind deleting this?

**Comment by pocket7878** [comment]
@DarkDimius @jez 
Thank you for your reviews!
I just updated my code!

**Comment by DarkDimius** [comment]
Thank you! Kicking off CI, will merge when it passes!
Thank you so much!

**Pull request merged**